src/lib/KeyStore.js

```diff
@@ -225,52 +240,63 @@ class KeyStore {
         }
     }
 
-    // eslint-disable-next-line valid-jsdoc
     /**
-     * @param {(AccountInfo|AccountRecord)[]} accounts
-     * @param {boolean} [withAccount]
-     * @returns {KeyguardRequest.KeyInfoObject[]|KeyguardRequest.LegacyKeyInfoObject[]|KeyRecord[]}
+     * @param {AccountRecord[]} accounts
+     * @returns {KeyRecord[]}
      */
-    static accounts2Keys(accounts, withAccount) {
+    static _accountRecords2KeyRecords(accounts) {
         return accounts.map(account => {
             const address = Nimiq.Address.fromUserFriendlyAddress(account.userFriendlyAddress);
-            const legacyKeyId = Key.deriveId(address.serialize());
+            const legacyKeyHash = Key.deriveHash(address.serialize());
+
+            return {
+                hash: legacyKeyHash,
+                type: Nimiq.Secret.Type.PRIVATE_KEY,
+                hasPin: account.type === 'low',
+                secret: account.encryptedKeyPair,
+            };
+        });
+    }
+
+    /**
+     * @param {AccountInfo[]} accounts
+     * @returns {KeyguardRequest.LegacyKeyInfoObject[]}
+     */
+    static accountInfos2KeyInfos(accounts) {
+        return accounts.map((account, index) => {
+            const address = Nimiq.Address.fromUserFriendlyAddress(account.userFriendlyAddress);
 
-            /** @type {KeyguardRequest.KeyInfoObject} */
-            const keyObject = {
-                id: legacyKeyId,
-                type: 1, // Nimiq.Secret.Type.PRIVATE_KEY
+            /** @type {KeyguardRequest.LegacyKeyInfoObject} */
+            const legacyKeyObject = {
+                id: index + 1,
```

Comment:
Ok, I actually think forecasting the legacy accounts' `id`s like that is fine now, because even on Safari the migration will happen as soon as a user tries to create or import another account (because that is a top-level invocation of the Keyguard). So I think everything is fine here and this _should_ work nicely.